opentelemetry-sdk/src/opentelemetry/sdk/util/__init__.py

```diff
@@ -148,3 +151,46 @@ def from_map(cls, maxlen, mapping):
         for key, value in mapping.items():
             bounded_dict[key] = value
         return bounded_dict
+
+
+class _Timer:
+    def __init__(self, histogram):
+        super().__init__()
+        self._start = None
+        self._attributes = None
+        self._histogram = histogram
+
+    def __enter__(self):
+        self._start = default_timer()
+        return self
+
+    def __exit__(
+        self,
+        exc_type: Optional[Type[BaseException]],
+        exc_val: Optional[BaseException],
+        exc_tb: Optional[TracebackType],
+    ) -> None:
+        duration = max(round((default_timer() - self._start) * 1000), 0)
+        self._histogram.record(duration, self._attributes)
+
+    def time(self, attributes=None):
+        self._attributes = attributes
+        return self
+
+    def set_attributes(self, attributes):
+        if self._attributes is None:
+            self._attributes = attributes
+        else:
+            self._attributes.update(attributes)
+
+    def __call__(self, func):
+        @wraps(func)
+        def wrapped(*args, **kwargs):
+            with self.time(attributes=self._attributes):
+                return func(*args, **kwargs)
+
+        return wrapped
+
+
+def get_timer(histogram):
```

Comment:
Does this mean that users must explicitly "convert" a histogram to a timing histogram to use it as a context manager?

Reply:
There is no timing histogram but a different object `_Timer` that behaves as a context manager that automatically adds timing information to an existing histogram. I pushed for this idea because:

1. The idea of having something that adds timing information to a histogram is outside of the spec, its something we (OTel Python) are introducing so it should be in a util library, not in the SDK itself.
2. Adding another API in the SDK would couple application code that uses this API with the current SDK, which should not happen implicitly (for example, by using another SDK and getting an attribute error because the other SDK does not have a `time` method in the `Histogram` class).